[FIX] sale: fix image link for onb. sample QO

flch-odoo opened the pull request:
The image for the sample product was updated in https://github.com/odoo/odoo/commit/76141b0dbd96717dd0bae62faa1a9318a5808a4f which made the
'Sample quotation' onboarding step hang for databases without any
`product.product` record.

A simple unit test for `_get_sample_sales_order` is included.

Technically, it wasn't possible to add it to the existing onboarding test
class because these tests are run post_install, and would fail on ENT builds
because of pos_loyalty preventing archiving `product.product`s.

Task-3045287

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/104221).

Feyensv reviewed (approved):
LGTM :)
Maybe worth making one little test going through some of those onboarding methods ?

flch-odoo commented:
Thank you @Feyensv !
They are tested through the tour, except this part because it didn't wait for the form view to be shown.
I fixed it too, and added another run simulating fresh installs sufficiently (fails without the link fix).

Demesmaeker commented:
Hello @flch-odoo 
About this tour you are changing, I feel like there are some little details that needs fine-tuning =]
First, this is also an onboarding tour. Since you didn't wrote any `content`, the user will see something like "click here to go to the next step", even if there are no next step.
Secondly, after this tour, you have another one called `sale_quote_tour` which starts right after `sale_tour` and whose first step is... triggering the `partner_id` as you did. In 15.2, this second tour isn't (yet) tested though, but it might be wiser to make your setup and use this tour instead. So: it'll be tested, and both tours won't collide with each other.
Apart from that, lgtm though =]

Feyensv commented on addons/sale_management/tests/test_sale_onboarding.py:
```diff
@@ -0,0 +1,14 @@
+
+
+class TestOnboarding(odoo.tests.HttpCase):
+    def test_01_get_sample_sales_order_from_scratch(self):
+        # Make sure there are no QO nor products
+        self.env['sale.order'].search([
+            ('company_id', '=', self.env.company.id),
+            ('partner_id', '=', self.env.user.partner_id.id),
+            ('state', '=', 'draft')
+        ]).state = 'cancel'
+        self.env['product.product'].search([]).active = False
+        self.env.company._get_sample_sales_order()
```
why didn't you specify this test in the `sale` module ?
Considering the method we want to test is there, the test should be too :thinking:

Feyensv commented on addons/sale/static/src/js/tours/sale.js:
```diff
@@ -53,6 +53,10 @@ tour.register("sale_tour", {
     extra_trigger: ".o_sale_order",
     content: _t("Now, we'll create a sample quote."),
     position: "bottom",
+}, {
+    trigger: "h1 span:contains('New')",
+    content: "Check we are on the right page",
+    run: function () {} // check
```
Considering it's the onboarding tour of sale, we should avoid adding 'technical' steps to it.
Remove this diff, the little onboarding test you added is already a good improvement for the coverage :).

Feyensv reviewed (commented):
minor test remarks :)

flch-odoo commented:
Sorry, @Feyensv , I had to remove the useless ENT commit linked. Then I just removed "_management" from the commit message. 
Would you please repeat the r+?

Feyensv commented:
No problem, thanks for noticing and improving the commit message :)

@robodoo r+
